pkg/security/probe/probe_kernel_file_windows.go

```diff
@@ -134,26 +132,26 @@ func parseCreateHandleArgs(e *etw.DDEventRecord) (*createHandleArgs, error) {
 	ca := &createHandleArgs{
 		DDEventHeader: e.EventHeader,
 	}
-	data := unsafe.Slice((*byte)(e.UserData), uint64(e.UserDataLength))
+	data := etwimpl.GetUserData(e)
 	if e.EventHeader.EventDescriptor.Version == 0 {
-		ca.irp = binary.LittleEndian.Uint64(data[0:8])
-		ca.threadID = binary.LittleEndian.Uint64(data[8:16])
-		ca.fileObject = fileObjectPointer(binary.LittleEndian.Uint64(data[16:24]))
-		ca.createOptions = binary.LittleEndian.Uint32(data[24:28])
-		ca.createAttributes = binary.LittleEndian.Uint32(data[28:32])
-		ca.shareAccess = binary.LittleEndian.Uint32(data[32:36])
-
-		ca.fileName, _, _, _ = etwutil.ParseUnicodeString(data, 36)
+		ca.irp = data.GetUint64(0)
+		ca.threadID = data.GetUint64(8)
+		ca.fileObject = fileObjectPointer(data.GetUint64(16))
+		ca.createOptions = data.GetUint32(24)
+		ca.createAttributes = data.GetUint32(28)
+		ca.shareAccess = data.GetUint32(32)
+
+		ca.fileName, _, _, _ = data.ParseUnicodeString(36)
 	} else if e.EventHeader.EventDescriptor.Version == 1 {
 
-		ca.irp = binary.LittleEndian.Uint64(data[0:8])
-		ca.fileObject = fileObjectPointer(binary.LittleEndian.Uint64(data[8:16]))
-		ca.threadID = uint64(binary.LittleEndian.Uint32(data[16:20]))
-		ca.createOptions = binary.LittleEndian.Uint32(data[20:24])
-		ca.createAttributes = binary.LittleEndian.Uint32(data[24:38])
-		ca.shareAccess = binary.LittleEndian.Uint32(data[28:32])
+		ca.irp = data.GetUint64(0)
+		ca.fileObject = fileObjectPointer(data.GetUint64(8))
+		ca.threadID = uint64(data.GetUint32(16))
+		ca.createOptions = data.GetUint32(20)
+		ca.createAttributes = data.GetUint32(24)
+		ca.shareAccess = data.GetUint32(28)
 
-		ca.fileName, _, _, _ = etwutil.ParseUnicodeString(data, 32)
+		ca.fileName, _, _, _ = data.ParseUnicodeString(32)
 	} else {
```

Comment:
this is where payoff of UserData is obvious 👍

Reply:
Interesting idea of UserData.GetXXX to return value and next offset would allow to write in the following way (not sure if it worth or worse it)

```
		ca.irp, off = data.GetUint64(off)
		ca.threadID, off = data.GetUint64(off)
		ca.fileObject, off = fileObjectPointer(data.GetUint64(off))
		ca.createOptions, off = data.GetUint32(off)
		ca.createAttributes, off = data.GetUint32(off)
		ca.shareAccess, off = data.GetUint32(off)

		ca.fileName, _, _, _ = data.ParseUnicodeString(off)
```
Alternatively current offset is inside of UserData

Reply:
that assumes that everything will be read, sequentially, nothing skipped, etc.

Reply:
Yes, but still it provides opportunity to enter what is needed. Another word is to pass pointer of cursor to be updated.